Take default GC choices always

gbaraldi opened the pull request:
Revert only the automatic tuning of the GC because there are some issues with it. @d-netto is looking into them but it's probably not going to be done for 1.10

gbaraldi committed:
Take default GC choices always

gbaraldi committed:
Fix unintended change

gbaraldi commented:
@d-netto can you check if this doesn't make the performance too much worse?

gbaraldi committed:
Move membalancer to an ifdef

gbaraldi committed:
Don't use square root logic because there are questions about it as well

gbaraldi committed:
Move allocation to ratio to a number more similar to 1.9

NHDaly commented on src/gc.c:
```diff
@@ -3421,9 +3427,17 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
         thrashing = 0; // maybe we should report this to the user or error out?
 
     int bad_result = (target_allocs*min_interval + heap_size) > 2 * jl_atomic_load_relaxed(&gc_heap_stats.heap_target); // Don't follow through on a bad decision
-    if (target_allocs == 0.0 || thrashing || bad_result) // If we are thrashing go back to default
-        target_allocs = 2*sqrt((double)heap_size/min_interval);
-    uint64_t target_heap = (uint64_t)target_allocs*min_interval + heap_size;
+    if (target_allocs == 0.0 || thrashing || bad_result || /*Always do the default to avoid issues with the algorithm*/ 1) // If we are thrashing go back to default
+        target_allocs =  2*sqrt((double)heap_size/min_interval);
```
In the MEMBALANCER case, this looks wrong: you're now missing the `*min_interval` multiplication on line 3440, right?
Should you have `target_heap *= min_interval` here?

gbaraldi replied:
Right, thanks!

gbaraldi committed:
Fix MEMBALANCER code

d-netto commented:
We [backported](https://github.com/RelationalAI/julia/pull/89) this PR and ran our internal CI with it.

There seems to be OOM kills in a couple of tests.

gbaraldi committed:
Add GOGC like variable to control GC frequency

gbaraldi committed:
Add null check

d-netto commented:
FWIW I can reproduce the issue reported on https://github.com/JuliaLang/julia/issues/51601 with this PR.

Peak RSS on 1.9 is around 400MB-500MB, and with this PR it goes beyond 60GB and fails on the serialization step of GCBenchmarks.

gbaraldi committed:
Merge branch 'master' into gb/gc-rever

gbaraldi committed:
Add the batch functions
+ adjust the target heap if too much gets freed out of GC
+ add the counted realloc size to the old gen

gbaraldi committed:
Whitespace

vtjnash commented:
I don't know if you wanted to pull out the `jl_batch_accum_heap_size` code reorg but sounds like this wasn't sufficient and so didn't turn out to be useful
